### DISCUSSION TITLE
Adding more context to the search result

### DISCUSSION THREAD

**Discussion by dprothero** [Pagefind/pagefind#119] [Ideas]
It would be nice if there where a way to add more context to a search result, such as where it was found within the hierarchy of the site's information architecture. See this mockup for an example:

![image](https://user-images.githubusercontent.com/1897006/198061156-01366cd0-fa25-477a-ae18-32e4395d05b4.png)

I suppose I could implement this by building my own UI and just using the API directly, but I'd thought I'd see if there was broader interest in something like this. Not sure how this would be implemented.

**Comment by bglw** [comment]
There's some semblance of this already with the tagged metadata, for instance if I add `data-pagefind-meta="section:Searching"` to a page of the Pagefind documentation, I'll see the following when searching:

<img width="753" alt="Screen Shot 2022-10-27 at 11 31 40 AM" src="https://user-images.githubusercontent.com/40188355/198151005-0b7cac17-35d8-4824-8709-20b1cdd5dcc5.png">

With the tagged section metadata. So the idea would be if you have the text `Home > REST API Gateway > FAQ` somewhere on the page, you could tag it as metadata and have it appear.

Beyond that, it comes down to the placement / styling of it to match the location in your mockup. Some amount of it could be achieved with some custom CSS. The remaining challenge would be moving that metadata above the title.

Perhaps similarly to how the UI looks for some special metadata keys (`title`, `image`, `url`), we could name another special key (`section` or `label`?) that gets picked up and rendered in the location of your mockup (and thus automatically hidden from the metadata pills below the excerpt).

**Comment by dprothero** [comment]
That could actually work just fine. It wasn't clear from reading the docs that's what I would get by using the metadata, but now that you illustrate it, I think that's the way to go, at least from the data perspective. UI preferences aside, I think this gets me most of the way there.

**Reply by bglw** [reply]
Sweet as, I'll continue pondering about the best way going forward to support peoples preferences in the UI 🙂
